### PR TITLE
feat: [DEL-3180]: clean ng token backend logic

### PR DESCRIPTION
**Pull request by arpit3107**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/30281)

### CODE REVIEW

File: 920-delegate-service-beans/src/main/java/io/harness/delegate/beans/DelegateTokenDetails.java [hunk 1]
```diff
@@ -25,6 +25,5 @@
   private long createdAt;
   private DelegateTokenStatus status;
   private String value;
-  private String identifier;
   private String ownerIdentifier;
```
Comment (abhinav-harness) [inline 920-delegate-service-beans/src/main/java/io/harness/delegate/beans/DelegateTokenDetails.java:28]: cant remove as we already have some mongo records

Comment (arpit3107) [reply 920-delegate-service-beans/src/main/java/io/harness/delegate/beans/DelegateTokenDetails.java:28]: I checked in mongo, there are no records which has identifier field.

**Review by abhinav-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30281)* status: 18 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java, line 493 at r3](https://reviewable.io/reviews/harness/harness-core/30281#-MuRzzLK8_bHBNsBXOmm:-MuRzzLK8_bHBNsBXOmn:b-1ciz2) ([raw file](https://github.com/harness/harness-core/blob/07ac08957fb1828d56d58399fe08dda82798c2f5/400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java#L493)):*
> ```Java
>     }
> 
>     delegateNgTokenService.upsertDefaultToken(account.getUuid(), null, false);
> ```

dont we need it?

**Review by arpit3107 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30281)* status: 18 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 1 approvals obtained. 

___
*[400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java, line 493 at r3](https://reviewable.io/reviews/harness/harness-core/30281#-MuRzzLK8_bHBNsBXOmm:-MuUAhGC7TEvodgc3HCA:b-908we) ([raw file](https://github.com/harness/harness-core/blob/07ac08957fb1828d56d58399fe08dda82798c2f5/400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java#L493)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

dont we need it?
</blockquote></details>

No. instead i have implemented AccountCrudObservers in DelegateNgTokenServiceImpl.


  @Override
  public void onAccountCreated(Account account) {
    upsertDefaultToken(account.getUuid(), null, false);
  }

**Review by abhinav-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30281)* status: 19 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[400-rest/src/main/java/io/harness/service/impl/DelegateNgTokenServiceImpl.java, line 76 at r4](https://reviewable.io/reviews/harness/harness-core/30281#-MuVO-zgC9BySCtPLVZ_:-MuVO-zgC9BySCtPLVZa:b-1v5shd) ([raw file](https://github.com/harness/harness-core/blob/ace018ca843ddaf04189fc68eea327462c94e732/400-rest/src/main/java/io/harness/service/impl/DelegateNgTokenServiceImpl.java#L76)):*
> ```Java
>                                       .value(encodeBase64(Misc.generateSecretKey()))
>                                       .createdAt(System.currentTimeMillis())
>                                       .createdBy(getEmbeddedUser())
> ```

taken care by morphia itself



_Code quote:_
```Java
                                      .createdAt(System.currentTimeMillis())
                                      .createdBy(getEmbeddedUser())
```
___
*[400-rest/src/main/java/io/harness/service/impl/DelegateNgTokenServiceImpl.java, line 110 at r4](https://reviewable.io/reviews/harness/harness-core/30281#-MuVP09o1bm93aP6TbFs:-MuVP09o1bm93aP6TbFt:b8z36ty) ([raw file](https://github.com/harness/harness-core/blob/ace018ca843ddaf04189fc68eea327462c94e732/400-rest/src/main/java/io/harness/service/impl/DelegateNgTokenServiceImpl.java#L110)):*
> ```Java
>     }
>     return query.asList().stream().map(token -> getDelegateTokenDetails(token, false)).collect(Collectors.toList());
>   }
> ```

do pagination later
___
*[400-rest/src/main/java/io/harness/service/impl/DelegateNgTokenServiceImpl.java, line 237 at r4](https://reviewable.io/reviews/harness/harness-core/30281#-MuVNwFW6xrC6M7EEMAw:-MuVNwFW6xrC6M7EEMAx:b-7ixkdm) ([raw file](https://github.com/harness/harness-core/blob/ace018ca843ddaf04189fc68eea327462c94e732/400-rest/src/main/java/io/harness/service/impl/DelegateNgTokenServiceImpl.java#L237)):*
> ```Java
>   }
> 
>   private EmbeddedUser getEmbeddedUser() {
> ```

not req, pl test



_Code quote:_
```Java
e
```
___
*[400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java, line 493 at r4](https://reviewable.io/reviews/harness/harness-core/30281#-MuVNBxSEKA5u7ThrVTR:-MuVNBxSEKA5u7ThrVTS:b-t4p9vm) ([raw file](https://github.com/harness/harness-core/blob/07ac08957fb1828d56d58399fe08dda82798c2f5/400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java#L493)):*
> plemented AccountCrudObservers in DelegateNg

**Comment by raghuAtWings** [comment]
___
*[360-cg-manager/src/main/java/software/wings/app/WingsApplication.java, line 151 at r5](https://reviewable.io/reviews/harness/harness-core/30281#-MuWsV8a4JvqqNrA5jbQ:-MuWsV8a4JvqqNrA5jbR:bblyvqy) ([raw file](https://github.com/harness/harness-core/blob/9c96f50ed3c00f11983bd29b5c652be867971c5c/360-cg-manager/src/main/java/software/wings/app/WingsApplication.java#L151)):*
> ```Java
> import io.harness.service.impl.DelegateInsightsServiceImpl;
> import io.harness.service.impl.DelegateSyncServiceImpl;
> import io.harness.service.impl.DelegateTaskServiceImpl;
> ```

you would need to fix these imports

**Comment by raghuAtWings** [comment]
___
*[400-rest/src/main/java/io/harness/service/impl/DelegateNgTokenServiceImpl.java, line 75 at r5](https://reviewable.io/reviews/harness/harness-core/30281#-MuWslTJ4rNFKcW4f1Fe:-MuWslTJ4rNFKcW4f1Ff:bbvgfhr) ([raw file](https://github.com/harness/harness-core/blob/ae46c30385cef61591760c5925b2f7757c38c0e6/400-rest/src/main/java/io/harness/service/impl/DelegateNgTokenServiceImpl.java#L75)):*
> ```Java
> 
>     persistence.save(delegateToken);
>     // publishCreateTokenAuditEvent(delegateToken);
> ```

why audit event is commented?

**Comment by raghuAtWings** [comment]
___
*[400-rest/src/main/java/io/harness/service/impl/DelegateNgTokenServiceImpl.java, line 125 at r5](https://reviewable.io/reviews/harness/harness-core/30281#-MuWsx4jAI_hSRSdbwim:-MuWsx4jAI_hSRSdbwin:bpkt9kr) ([raw file](https://github.com/harness/harness-core/blob/ae46c30385cef61591760c5925b2f7757c38c0e6/400-rest/src/main/java/io/harness/service/impl/DelegateNgTokenServiceImpl.java#L125)):*
> ```Java
>                                                     .findFirst()
>                                                     .orElse(null);
>     return delegateTokenDetails != null ? decodeBase64ToString(delegateTokenDetails.getValue()) : null;
> ```

i don't think we are getting any value from base64 encoding,

**Comment by raghuAtWings** [comment]
___
*[400-rest/src/main/java/io/harness/service/impl/DelegateNgTokenServiceImpl.java, line 139 at r5](https://reviewable.io/reviews/harness/harness-core/30281#-MuWtIUUE5kaYlwT7NKY:-MuWtIUUE5kaYlwT7NKZ:b-hh9zhn) ([raw file](https://github.com/harness/harness-core/blob/ae46c30385cef61591760c5925b2f7757c38c0e6/400-rest/src/main/java/io/harness/service/impl/DelegateNgTokenServiceImpl.java#L139)):*
> ```Java
> 
>     Query<DelegateToken> queryExistsActive = query.filter(DelegateTokenKeys.status, DelegateTokenStatus.ACTIVE);
>     Optional<DelegateToken> token = queryExistsActive.asList().stream().findAny();
> ```

we can just do queryExistsActive.get()

**Review by raghuAtWings — COMMENTED** [review]
> Reviewed 5 of 9 files at r1, 11 of 13 files at r2, 1 of 1 files at r3, 1 of 1 files at r4, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30281)* status: 1 change requests. 0 of 1 approvals obtained.

**Pull request closed without merging**